cosmos/_triggers/watcher.py

```diff
@@ -87,23 +87,21 @@ async def get_xcom_val(self, key: str) -> Any | None:
             return await self.get_xcom_val_af3(key)
 
     async def _parse_node_status(self) -> str | None:
-        key = f"nodefinished_{self.model_unique_id.replace('.', '__')}" if self.use_event else "run_results"
-
-        compressed_xcom_val = await self.get_xcom_val(key)
-        if not compressed_xcom_val:
-            return None
-
-        data_json = _parse_compressed_xcom(compressed_xcom_val)
+        key = (
+            f"nodefinished_{self.model_unique_id.replace('.', '__')}"
+            if self.use_event
+            else f"{self.model_unique_id.replace('.', '__')}_status"
+        )
 
         if self.use_event:
+            compressed_xcom_val = await self.get_xcom_val(key)
+            if not compressed_xcom_val:
+                return None
+
+            data_json = _parse_compressed_xcom(compressed_xcom_val)
             return data_json.get("data", {}).get("run_result", {}).get("status")  # type: ignore[no-any-return]
 
-        results = data_json.get("results", [])
-        node_result: dict[str, Any] = next(
-            (r for r in results if r.get("unique_id") == self.model_unique_id),
-            {},
-        )
-        return node_result.get("status")
```

Comment:
Why did we remove this part of the code?

Reply:
This parsing was performed to retrieve the status of a dbt resource from the run_result. Now, dbt resource state is available in xcom, so we don't need parsing